docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +275,117 @@ Note that:
+  * It must subscribe to `tedge/$CHILD_DEVICE_ID/commands/req/#` to receive requests.
+  * It has to publish responses under `tedge/$CHILD_DEVICE_ID/commands/res/#`.
+* On demand, the child-device software has to send HTTP requests to thin-edge.
+  * These requests are REST requests (GET/PUT/DELETE) to exchange files.
+  * The urls are forged by thin-edge telling the child devices
+    where to get and put configuration files.
+* Currently, all these connections are done without TLS support.
+  * Hence, the main device must be configured so these MQTT and HTTP ports
+    are only open on a local trusted network.
+
+### The child device downloads configuration updates on notification
+
+When a configuration file for a child device
+is received from the cloud by the `c8y_configuration_plugin` daemon,
+the configuration plugin manages the transfer of this file to the child device
+and notifies the cloud on the progress of this configuration update operation.
+
+1. On reception of a configuration update for a child device named `$CHILD_DEVICE_ID`
+   of a file `{ path = $PATH, type = $TYPE }`defined in `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` 
+   the `c8y_configuration_plugin` downloads the new configuration content in `/tmp`,
+   and, on success, moves this content under the thin-edge file transfer HTTP root.
+   * The file is moved to `$TEDGE_HTTP_ROOT/config_snapshot/$CHILD_DEVICE_ID/$TYPE`
+     making this file available under `http://$TEDGE_HTTP/tedge/config_snapshot/$CHILD_DEVICE_ID/$TYPE`.
+   * Note that the `$TYPE` is by default the `$PATH` if not provided by the configuration.  
+1. Once the updated configuration is available over the local HTTP file transfer service,
+   the `c8y_configuration_plugin` notify the child device by publishing an MQTT message.
+   * The topic is `tedge/$CHILD_DEVICE_ID/commands/req/config_update`
+   * The payload is a JSON record with 3 fields
+     * `"url": "http://$TEDGE_HTTP/tedge/config_snapshot/$CHILD_DEVICE_ID/$PATH"`
+     * `"path": "$PATH"`
+     * `"type": "$TYPE"` (if no `type` has been specified, then this field is omitted)
+1. On reception of a configuration update on the topic `tedge/$CHILD_DEVICE_ID/commands/req/config_update`,
+   The child-device specific software for configuration management:
+   1. Downloads the content from the `url` specified by the notification message.
+   1. Uses the `path` and `type` information to apply the new configuration content.
+      Note that these pieces of information are provided by the child-device specific software itself,
+      and make sense only in the specific context of the device operating system and software.
+1. While the configuration update is applied,
+   the child-device specific software for configuration management,
+   notifies thin-edge over MQTT about the progress of this operation.
+   1. These messages are published on the topic
+      `tedge/$CHILD_DEVICE_ID/commands/res/config_update`/<config file path>`
```

Comment:
Format broken.
```suggestion
      `tedge/$CHILD_DEVICE_ID/commands/res/config_update/<config file path>`
```

Reply:
Oh wait, do we really need the last `<config file path>`? I would omit it, and make the topic `tedge/$CHILD_DEVICE_ID/commands/res/config_update`. Two reasons:
1. `type` and `path` will be send by payload.
2. I checked how tedge_agent sends status message to mapper. [Here](https://github.com/thin-edge/thin-edge.io-specs/blob/main/src/software-management/sm-agent.md#thin-edge-json-software-update-response). The topic is `tedge/commands/res/software/update`. Without `<config file path>, it look more aligned with our existing schema.

Reply:
Oops. You are correct and it was my intent. I will fix that.

Reply:
Fixed